97_extra_materials.Rmd

```diff
@@ -290,3 +290,324 @@ e <- attr(reducedDim(tse_Genus, "PCoA_BC"), "eig");
+# Inspect the new colData with added alpha diversity estimate
+colData(TSE) %>% names # shannon has been added to the colData
+
+```
+
+If we want to extract a data frame that only has the alpha diversity it can be done easily.
+
+```{r}
+# Extract
+df <- colData(TSE) %>% data.frame %>% dplyr::select(matches("shannon"))
+
+```
+
+
+### Calculating beta diversity: ordinate = runMDS
+
+We can calculate PcoA with Bray-Curtis distances in Phyloseq using the ordiate command. This 
+The beta diversity calculation in mia  outputs a TreeSE with a new type of data, reduced dimensions or reducedDim. Here we will use the scater package that runs the PcoA with runMDS. (PcoA and MDS mean the same thing)
+
+In Phyloseq you would again need to add the dimensions to the sample data if you want to keep them safe with other metadata.
+
+```{r}
+# Run PCoA on the relative abundance data and store in PHY_ord list
+PHY_ord <- ordinate(PHY_relab, method = "PCoA", distance = "bray")
+
+library(scater)
+
+# For the ordination using mia we use runMDDS from the scater package and implement the vegan bray distance calculation
+TSE <- runMDS(TSE,
+              FUN = vegan::vegdist,
+              method = "bray",
+              assay.type = "relabundance",
+              name = "MDS_bray",
+              ncomponents = 10) # Let's also define how many dimensions
+TSE # Inspect, now we have new reducedDim "MDS_bray"
+
+```
+
+### Plotting ordinations: plot_ordination = plotReducedDim
+Phyloseq has it's own plotting fuction for ordinations.
+
+```{r}
+plot_ordination(physeq = PHY, ordination = PHY_ord, color = "SampleType")
+```
+
+It is also easy to plot the ordination stored in reducedDim in the TSE using the plotReducedDim. We can first check what the name of the Bray-Curtis MDS/PCoA was.
+```{r}
+# Check reduced dim names
+reducedDimNames(TSE)
+
+```
+Ok, let's plot.
+```{r}
+# Plot
+plotReducedDim(TSE, "MDS_bray", color_by = "SampleType")
+
+# The sign is given arbitrarily so we can change it to match the plot_ordination
+
+reducedDim(TSE)[,1] <- -reducedDim(TSE)[,1]
+reducedDim(TSE)[,2] <- -reducedDim(TSE)[,2]
+
+plotReducedDim(TSE, "MDS_bray", color_by = "SampleType")
+```
+
+
+### Agglomerating taxa: tax_glom=mergeFeaturesByRank
+
+Often you might want to study your data using different taxonomic ranks, for example check if you see differences in the abundances of higher taxonomic levels.
+
+This data object can again be conveniently stored in a TSE object under altExp.
+
+
+
+```{r}
+PHY_fam <- tax_glom(PHY, taxrank = "Family")
+
+
+```
+
+Tax glom removes the taxa which have not been assigned to the level given in taxrank by default. NArm = TRUE so we will add the onRankOnly = TRUE which is equivalent to the mergeFeaturesByRank.
+
+```{r}
+altExp(TSE, "Family") <-
+  mergeFeaturesByRank(TSE,
+                      rank = "Family",
+                      onRankOnly = TRUE, # Let's add the equivalent option to NArm =TRUE
+                      agglomerateTree = TRUE)
```

Comment:
This seems to be the default, so should we leave it out for simplicity?

Reply:
onRankOnly is FALSE by default?